src/main/java/htsjdk/samtools/util/BufferedLineReader.java

```diff
@@ -54,7 +76,18 @@ public BufferedLineReader(final InputStream is, final int bufferSize) {
      * is necessary because the String is in unicode.
      */
     public static BufferedLineReader fromString(final String s) {
-        return new BufferedLineReader(new ByteArrayInputStream(s.getBytes()));
+        final InputStream is;
+        // Developer Note: if the string is long enough such that the # of bytes needed exceeds the
+        // maximum array size, we need to use a custom string-backed input stream, versus a byte array
+        // backed input stream.  The # of bytes needed is the length of the string times the number
+        // bytes per character to store in UTF8.
+        if (s.length() * MAX_BYTES_PER_CHAR_UTF8 >= Integer.MAX_VALUE) {
```

Comment:
@pshapiro4broad I think this solves the issue, and I added comments to describe what we're doing here. Can you look one more time?